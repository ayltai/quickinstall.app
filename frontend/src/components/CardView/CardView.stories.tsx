import type { Meta, StoryObj, } from '@storybook/react';
import React from 'react';

import { MaterialButton, } from '../Button';
import { CardView, } from './CardView';

const meta : Meta<typeof CardView> = {
    component : CardView,
    title     : 'Components/Data Presentations/Card',
    argTypes  : {
        title       : {
            description : 'Title of the card',
        },
        subtitle    : {
            description : 'Subtitle of the card',
        },
        description : {
            description : 'Description of the card',
        },
        onClick     : {
            description : 'Function to call when the button is clicked',
        },
        children    : {
            description : 'One or more buttons',
        },
    },
};

export default meta;

type Story = StoryObj<typeof CardView>;

export const Default : Story = {
    args : {
        title       : 'Headline',
        subtitle    : 'Subheader',
        description : 'Explain more about the topic shown in the headline and subheader through supporting text.',
    },
};

export const WithButtons : Story = {
    args : {
        ...Default.args,
        children : [
            <MaterialButton
                key='1'
                variant='outlined'>
                Action 1
            </MaterialButton>,
            <MaterialButton
                key='2'
                variant='contained'>
                Action 2
            </MaterialButton>,
        ],
    },
};
