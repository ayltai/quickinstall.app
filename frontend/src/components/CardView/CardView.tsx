import Box from '@mui/material/Box';
import Card from '@mui/material/Card';
import CardActions from '@mui/material/CardActions';
import CardActionArea from '@mui/material/CardActionArea';
import CardContent from '@mui/material/CardContent';
import Tooltip from '@mui/material/Tooltip';
import React, { type FC, } from 'react';

import { TruncatedTypography, } from '../TruncatedTypography';
import type { CardViewProps, } from './CardView.types';

/**
 * A card is a container that can displays a title, subtitle, description, and optionally one or more buttons.
 * @param title Title of the card
 * @param subtitle Subtitle of the card
 * @param description Description of the card
 * @param onClick Function to call when the button is clicked
 * @param children One or more buttons
 * @param rest Other props
 */
export const CardView : FC<CardViewProps> = ({
    title,
    subtitle,
    description,
    children,
    onClick,
    ...rest
}) => (
    <Card {...rest}>
        <CardActionArea onClick={onClick}>
            <CardContent>
                <Tooltip title={title}>
                    <TruncatedTypography variant='h5'>
                        {title}
                    </TruncatedTypography>
                </Tooltip>
                <TruncatedTypography
                    variant='body2'
                    color='text.secondary'>
                    {subtitle}
                </TruncatedTypography>
                <Tooltip title={description}>
                    <TruncatedTypography
                        height='2lh'
                        lines={2}>
                        {description}
                    </TruncatedTypography>
                </Tooltip>
            </CardContent>
        </CardActionArea>
        {children !== undefined && (
            <CardActions>
                <>
                    <Box flexGrow={1} />
                    {children}
                </>
            </CardActions>
        )}
    </Card>
);
