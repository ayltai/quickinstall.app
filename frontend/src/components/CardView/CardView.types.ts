import type { HTMLAttributes, ReactNode, } from 'react';

export type CardViewProps = HTMLAttributes<HTMLDivElement> & {
    title?       : string,
    subtitle?    : string,
    description? : string,
    onClick?     : () => void,
    children?    : ReactNode,
};
