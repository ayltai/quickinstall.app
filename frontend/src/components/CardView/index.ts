export { CardView, } from './CardView';
export { SelectableCardView, } from './SelectableCardView';
export type { SelectableCardViewProps, } from './SelectableCardView.types';
