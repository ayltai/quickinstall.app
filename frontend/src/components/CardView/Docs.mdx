import { Controls, Description, Meta, Title, Primary, Stories, } from '@storybook/blocks';

import * as ComponentStories from './CardView.stories';

<Meta of={ComponentStories} />

<Title />

<Description />

# How to use cards

Cards can be used for many different purposes, including product or service tiles. They help users to scan information by visually grouping related content together.

When using cards,
* Use a descriptive title so that users can easily find and browse item
* Make sure all the content in the card, like images and action buttons, are contextually related
* Keep the number of action buttons per card to a minimum

# Default usage

<Primary />

# Parameters

<Controls sort='requiredFirst' />

<Stories
    title='Variants of buttons'
    includePrimary={false} />
