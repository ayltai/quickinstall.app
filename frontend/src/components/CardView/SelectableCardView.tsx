import Check from '@mui/icons-material/Check';
import DeleteForever from '@mui/icons-material/DeleteForever';
import { useTheme, } from '@mui/material/styles';
import React, { type ComponentType, JSX, useMemo, useState, } from 'react';
import { useTranslation, } from 'react-i18next';

import { ToggleButton, } from '../ToggleButton';
import { DialogView, } from '../DialogView';
import { MaterialButton, } from '../Button';
import { CardView, } from './CardView';
import type { CardViewProps, } from './CardView.types';
import type { SelectableCardViewProps, } from './SelectableCardView.types';

const withSelectable = (Component : ComponentType<CardViewProps>) => {
    const ComponentWithSelectable : (props : SelectableCardViewProps) => JSX.Element = (props : SelectableCardViewProps) => {
        const { style, selected = false, color, dialogAction, actionIcon, actionText, children, onChange, ...rest } = props;

        const [ open, setOpen, ] = useState(false);

        const theme = useTheme();

        const { t, } = useTranslation();

        const borderColor = useMemo(() => color === 'info' ? theme.palette.info.main : color === 'primary' ? theme.palette.primary.main : color === 'secondary' ? theme.palette.secondary.main : theme.palette.divider, [ color, theme, ]);

        const handleClick = () => setOpen(true);

        const handleChange = (checked : boolean) => onChange && onChange(checked);

        const handleClose = () => setOpen(false);

        return (
            <>
                <Component
                    style={{
                        ...style,
                        border          : `1px solid ${selected ? borderColor : 'transparent'}`,
                        backgroundColor : selected ? theme.palette.action.selected : theme.palette.background.paper,
                    }}
                    onClick={handleClick}
                    {...rest}>
                    {(actionIcon || actionText) && (
                        <MaterialButton
                            color={color}
                            startIcon={actionIcon}
                            onClick={handleClick}>
                            {actionText}
                        </MaterialButton>
                    )}
                    <ToggleButton
                        selected={selected}
                        color={color}
                        icon={<Check />}
                        text={t('action.select')}
                        selectedIcon={<Check />}
                        selectedText={t('action.deselect')}
                        deselectIcon={<DeleteForever />}
                        deselectText={t('action.remove')}
                        onChange={handleChange} />
                </Component>
                {children && (
                    <DialogView
                        open={open}
                        color={color}
                        title={props.title}
                        action={dialogAction}
                        onClose={handleClose}>
                        {children}
                    </DialogView>
                )}
            </>
        );
    };

    return ComponentWithSelectable;
};

/**
 * A selectable card displays a title, subtitle, description, and optionally one or more buttons.
 */
export const SelectableCardView = withSelectable(CardView);
