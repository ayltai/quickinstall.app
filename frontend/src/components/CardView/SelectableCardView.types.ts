import { ReactNode, } from 'react';

import { CardViewProps, } from './CardView.types';

export type SelectableCardViewProps = Omit<CardViewProps, 'children' | 'onChange'> & {
    selected?     : boolean,
    color?        : 'inherit' | 'primary' | 'secondary' | 'success' | 'error' | 'info' | 'warning',
    actionText?   : string,
    actionIcon?   : ReactNode,
    dialogAction? : ReactNode,
    children?     : ReactNode,
    onChange?     : (selected : boolean) => void,
};
