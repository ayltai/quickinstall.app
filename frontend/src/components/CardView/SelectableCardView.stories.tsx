import { Settings, } from '@mui/icons-material';
import { Box, Link, Stack, Typography, } from '@mui/material';
import type { Meta, StoryObj, } from '@storybook/react';
import React, { useState, } from 'react';

import { SelectableCardView, } from './SelectableCardView';

const meta : Meta<typeof SelectableCardView> = {
    component : SelectableCardView,
    title     : 'Components/Data Presentations/Selectable Card',
    tags      : [
        'autodocs',
    ],
    argTypes  : {
        selected     : {
            description : 'Whether the card is selected.',
        },
        color        : {
            description : 'Color of the card. Defaults to `inherit`.',
        },
        dialogAction : {
            description : 'The primary action to show in the action area.',
        },
        actionText   : {
            description : 'Text to show in the action area.',
        },
        actionIcon   : {
            description : 'Icon to show in the action area.',
        },
        children     : {
            description : 'The detailed content of the component',
        },
        onChange     : {
            description : 'Function to call when the card selection state is changed.',
        },
    },
};

export default meta;

type Story = StoryObj<typeof SelectableCardView>;

const description : string = 'Explain more about the topic shown in the headline and subheader through supporting text.';
const infoUrl     : string = 'https://nodejs.org';

export const Default : Story = {
    args   : {
        description,
        color    : 'primary',
        title    : 'Headline',
        subtitle : 'Subheader',
        children : (
            <Stack spacing={2}>
                <Box>
                    <Typography
                        variant='h5'
                        fontWeight='bold'>
                        Description
                    </Typography>
                    <Typography>
                        {description}
                    </Typography>
                </Box>
                <Box>
                    <Typography
                        variant='h5'
                        fontWeight='bold'>
                        Website
                    </Typography>
                    <Link
                        href={infoUrl}
                        target='_blank'>
                        <Typography>
                            {infoUrl}
                        </Typography>
                    </Link>
                </Box>
            </Stack>
        ),
    },
    render : props => {
        const [ selected, setSelected, ] = useState(false);

        return (
            <SelectableCardView
                {...props}
                selected={selected}
                onChange={setSelected} />
        );
    },
};

export const WithAction : Story = {
    args : {
        ...Default.args,
        actionText : 'Configure',
        actionIcon : <Settings />,
    },
};
