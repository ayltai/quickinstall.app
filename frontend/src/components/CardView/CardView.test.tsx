import React from 'react';

import { fireEvent, render, } from '../../utils/test';
import { MaterialButton, } from '../Button';
import { CardView, } from './CardView';

const TITLE       = 'Headline';
const SUBTITLE    = 'Subheader';
const DESCRIPTION = 'Explain more about the topic shown in the headline and subheader through supporting text.';

describe('<CardView />', () => {
    it('renders without buttons correctly', () => expect(render(
        <CardView
            title={TITLE}
            subtitle={SUBTITLE}
            description={DESCRIPTION} />
    )).toMatchSnapshot());

    it('renders with buttons correctly', () => expect(render(
        <CardView
            title={TITLE}
            subtitle={SUBTITLE}
            description={DESCRIPTION}>
            <MaterialButton key="1">Action 1</MaterialButton>
            <MaterialButton key="2">Action 2</MaterialButton>
        </CardView>
    )).toMatchSnapshot());

    it('triggers onClick when the card is clicked', () => {
        const handleClick = vi.fn();

        const { getByRole, } = render(
            <CardView
                title={TITLE}
                subtitle={SUBTITLE}
                description={DESCRIPTION}
                onClick={handleClick} />
        );

        fireEvent.click(getByRole('button'));

        expect(handleClick).toHaveBeenCalledTimes(1);
    });
});
