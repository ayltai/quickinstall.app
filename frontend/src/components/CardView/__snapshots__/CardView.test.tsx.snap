// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<CardView /> > renders with buttons correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-5f3tb7-MuiPaper-root-MuiCard-root"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root css-1oitmrd-MuiButtonBase-root-MuiCardActionArea-root"
          tabindex="0"
          type="button"
        >
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <h5
              aria-label="Headline"
              class="MuiTypography-root MuiTypography-h5  css-4lba41-MuiTypography-root"
              data-mui-internal-clone-element="true"
            >
              Headline
            </h5>
            <p
              class="MuiTypography-root MuiTypography-body2 css-sebv0w-MuiTypography-root"
            >
              Subheader
            </p>
            <p
              aria-label="Explain more about the topic shown in the headline and subheader through supporting text."
              class="MuiTypography-root MuiTypography-body1  css-gqj3pa-MuiTypography-root"
              data-mui-internal-clone-element="true"
              lines="2"
            >
              Explain more about the topic shown in the headline and subheader through supporting text.
            </p>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiCardActions-root MuiCardActions-spacing css-1vzhawx-MuiCardActions-root"
        >
          <div
            class="MuiBox-root css-i9gxme"
          />
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-lzklvy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Action 1
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-lzklvy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Action 2
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-5f3tb7-MuiPaper-root-MuiCard-root"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root css-1oitmrd-MuiButtonBase-root-MuiCardActionArea-root"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <h5
            aria-label="Headline"
            class="MuiTypography-root MuiTypography-h5  css-4lba41-MuiTypography-root"
            data-mui-internal-clone-element="true"
          >
            Headline
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body2 css-sebv0w-MuiTypography-root"
          >
            Subheader
          </p>
          <p
            aria-label="Explain more about the topic shown in the headline and subheader through supporting text."
            class="MuiTypography-root MuiTypography-body1  css-gqj3pa-MuiTypography-root"
            data-mui-internal-clone-element="true"
            lines="2"
          >
            Explain more about the topic shown in the headline and subheader through supporting text.
          </p>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiCardActions-root MuiCardActions-spacing css-1vzhawx-MuiCardActions-root"
      >
        <div
          class="MuiBox-root css-i9gxme"
        />
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-lzklvy-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Action 1
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-lzklvy-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Action 2
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<CardView /> > renders without buttons correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-5f3tb7-MuiPaper-root-MuiCard-root"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root css-1oitmrd-MuiButtonBase-root-MuiCardActionArea-root"
          tabindex="0"
          type="button"
        >
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <h5
              aria-label="Headline"
              class="MuiTypography-root MuiTypography-h5  css-4lba41-MuiTypography-root"
              data-mui-internal-clone-element="true"
            >
              Headline
            </h5>
            <p
              class="MuiTypography-root MuiTypography-body2 css-sebv0w-MuiTypography-root"
            >
              Subheader
            </p>
            <p
              aria-label="Explain more about the topic shown in the headline and subheader through supporting text."
              class="MuiTypography-root MuiTypography-body1  css-gqj3pa-MuiTypography-root"
              data-mui-internal-clone-element="true"
              lines="2"
            >
              Explain more about the topic shown in the headline and subheader through supporting text.
            </p>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-5f3tb7-MuiPaper-root-MuiCard-root"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root css-1oitmrd-MuiButtonBase-root-MuiCardActionArea-root"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <h5
            aria-label="Headline"
            class="MuiTypography-root MuiTypography-h5  css-4lba41-MuiTypography-root"
            data-mui-internal-clone-element="true"
          >
            Headline
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body2 css-sebv0w-MuiTypography-root"
          >
            Subheader
          </p>
          <p
            aria-label="Explain more about the topic shown in the headline and subheader through supporting text."
            class="MuiTypography-root MuiTypography-body1  css-gqj3pa-MuiTypography-root"
            data-mui-internal-clone-element="true"
            lines="2"
          >
            Explain more about the topic shown in the headline and subheader through supporting text.
          </p>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
